src/hotspot/share/classfile/classLoaderData.hpp

```diff
@@ -303,8 +305,8 @@ class ClassLoaderData : public CHeapObj<mtClass> {
   }
 
   // Used to refcount a non-strong hidden class's s CLD in order to indicate their aliveness.
-  void inc_keep_alive();
-  void dec_keep_alive();
+  void inc_keep_alive_ref_count();
```

Comment:
This comment is slightly misleading. We have two mechanisms to determine a CLDs aliveness. The first is this ref-counting mechanism, the other is the GC tracing. One can misinterpret this comment to mean that we only use the ref-count to determine the aliveness.

Could we tweak this comment to say something that the ref-count is used to *force* the aliveness of CLDs?

Reply:
How about:
    // Used to refcount a non-strong hidden class's CLD in order to force its aliveness during
    // loading, when gc tracing may not find this CLD alive through the holder.